### PR TITLE
Introduce new Api version 2019-08-01

### PR DESCRIPTION
**Pull request by naveedaz**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [X] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [X] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [X] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [x] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [X] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

File: specification/web/resource-manager/Microsoft.DomainRegistration/stable/2019-08-01/Domains.json [hunk 1]
```diff
@@ -0,0 +1,1477 @@
...
+          },
+          {
+            "$ref": "#/parameters/apiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/DomainAvailablilityCheckResult"
```
Comment (nschonni) [inline specification/web/resource-manager/Microsoft.DomainRegistration/stable/2019-08-01/Domains.json]:
```suggestion
              "$ref": "#/definitions/DomainAvailabilityCheckResult"
```

File: specification/web/resource-manager/Microsoft.DomainRegistration/stable/2019-08-01/Domains.json [hunk 2]
```diff
@@ -0,0 +1,1477 @@
...
+                "create",
+                "read"
+              ]
+            }
+          },
+          "x-ms-client-flatten": true
+        }
+      }
+    },
+    "DomainAvailablilityCheckResult": {
```
Comment (nschonni) [inline specification/web/resource-manager/Microsoft.DomainRegistration/stable/2019-08-01/Domains.json]:
```suggestion
    "DomainAvailabilityCheckResult": {
```

File: specification/web/resource-manager/Microsoft.Web/stable/2019-08-01/AppServiceEnvironments.json [hunk 3]
```diff
@@ -0,0 +1,2693 @@
...
+            "in": "query",
+            "description": "Specify <code>true</code> to include instance details. The default is <code>false</code>.",
+            "type": "boolean"
+          },
+          {
+            "name": "$filter",
+            "in": "query",
+            "description": "Return only usages/metrics specified in the filter. Filter conforms to odata syntax. Example: $filter=(name.value eq 'Metric1' or name.value eq 'Metric2') and startTime eq 2014-01-01T00:00:00Z and endTime eq 2014-12-31T23:59:59Z and timeGrain eq duration'[Hour|Minute|Day]'.",
+            "type": "string",
+            "x-ms-skip-url-encoding": t
```
Comment (nschonni) [inline specification/web/resource-manager/Microsoft.Web/stable/2019-08-01/AppServiceEnvironments.json]:
```suggestion
        "isAccessible": {
```

Comment (nschonni) [inline specification/web/resource-manager/Microsoft.Web/stable/2019-08-01/AppServiceEnvironments.json]:
```suggestion
        "diagnosticsOutput": {
```

Comment (nschonni) [inline specification/web/resource-manager/Microsoft.Web/stable/2019-08-01/AppServiceEnvironments.json]:
```suggestion
    "MetricAvailability": {
```

Comment (nschonni) [inline specification/web/resource-manager/Microsoft.Web/stable/2019-08-01/AppServiceEnvironments.json]:
```suggestion
                "$ref": "#/definitions/MetricAvailability"
```

File: specification/web/resource-manager/Microsoft.Web/stable/2019-08-01/AppServicePlans.json [hunk 4]
```diff
@@ -0,0 +1,1847 @@
...
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Web/serverfarms/{name}/metricdefinitions": {
+      "get": {
+        "tags": [
+          "AppServicePlans"
+        ],
+        "summary": "Get metrics that can be queried for an App Service plan, and their definitions.",
+        "description": "Description for Get metrics that can be queried for an App Service plan, and their definitions.",
+        "operationId": "AppServicePlans_ListMetricDefintions",
```
Comment (nschonni) [inline specification/web/resource-manager/Microsoft.Web/stable/2019-08-01/AppServicePlans.json]:
```suggestion
        "operationId": "AppServicePlans_ListMetricDefinitions",
```

File: specification/web/resource-manager/Microsoft.Web/stable/2019-08-01/CommonDefinitions.json [hunk 5]
```diff
@@ -0,0 +1,3072 @@
...
+          "type": "string"
+        },
+        "reroutePercentage": {
+          "format": "double",
+          "description": "Percentage of the traffic which will be redirected to <code>ActionHostName</code>.",
+          "type": "number"
+        },
+        "changeStep": {
+          "format": "double",
+          "description": "In auto ramp up scenario this is the step to add/remove from <code>ReroutePercentage</code> until it reaches \n<code>MinReroutePercentage</code> or <code>MaxReroutePercentage</code>. Site metrics are checked every N minutes specificed in <code>ChangeIntervalInMinutes</code>.\nCustom decision algorithm can be provided in TiPCallback site extension which URL can be specified in <code>ChangeDecisionCallbackUrl</code>.",
```
Comment (nschonni) [inline specification/web/resource-manager/Microsoft.Web/stable/2019-08-01/CommonDefinitions.json]:
```suggestion
          "description": "In auto ramp up scenario this is the step to add/remove from <code>ReroutePercentage</code> until it reaches \n<code>MinReroutePercentage</code> or <code>MaxReroutePercentage</code>. Site metrics are checked every N minutes specified in <code>ChangeIntervalInMinutes</code>.\nCustom decision algorithm can be provided in TiPCallback site extension which URL can be specified in <code>ChangeDecisionCallbackUrl</code>.",
```

File: specification/web/resource-manager/Microsoft.Web/stable/2019-08-01/AppServiceEnvironments.json (hunk 3 above)
Comment (naveedaz) [reply specification/web/resource-manager/Microsoft.Web/stable/2019-08-01/AppServiceEnvironments.json]: Fixed

**Comment by naveedaz** [comment]
@tjprescott Can you help figure out why SDK javascript and SDK python failed? SDK Java is still in progress, so not sure if it will fail too.

**Comment by naveedaz** [comment]
@tjprescott Nevermind. I believe I figured out all the issues. Please review.

**Review by KrisBash — APPROVED** [review]
Changes from prior version LGTM

**Review by tjprescott — APPROVED** [review]

**Pull request merged**